From 3de8991c85432f5c8f84ebef409bd5a921e763ac Mon Sep 17 00:00:00 2001
From: Werner Koch <wk@gnupg.org>
Date: Sat, 28 Sep 2019 12:03:43 +0200
Subject: [PATCH 1/6] ecc: Add a keygrip testcase for cv25519.

--

Signed-off-by: Werner Koch <wk@gnupg.org>
---
 tests/keygrip.c | 11 +++++++++++
 1 file changed, 11 insertions(+)

diff --git a/tests/keygrip.c b/tests/keygrip.c
index f775f7a1..d30ec392 100644
--- a/tests/keygrip.c
+++ b/tests/keygrip.c
@@ -189,6 +189,17 @@ static struct
       " ))",
       "\x9D\xB6\xC6\x4A\x38\x83\x0F\x49\x60\x70"
       "\x17\x89\x47\x55\x20\xBE\x8C\x82\x1F\x47"
+    },
+    { /* Cv25519 */
+      GCRY_PK_ECC,
+      "(public-key"
+      " (ecc"
+      " (curve Curve25519)(flags djb-tweak)"
+      " (q #40"
+      "     918C1733127F6BF2646FAE3D081A18AE77111C903B906310B077505EFFF12740#)"
+      " ))",
+      "\x0F\x89\xA5\x65\xD3\xEA\x18\x7C\xE8\x39"
+      "\x33\x23\x98\xF5\xD4\x80\x67\x7D\xF4\x9C"
     }
   };
 
-- 
2.25.0

